app/services/filter_tagged_content.rb

```diff
@@ -0,0 +1,62 @@
+class FilterTaggedContent
+  def valid?(document)
```

Comment:
I tend to avoid if/else clauses because I think there is a bigger chance of the code becoming too nested/hard to read over time. I personally find ruby's postfix conditionals a nice way of reading code.